mitiq/calibration/calibration.py

```diff
@@ -0,0 +1,204 @@
+            circuit = circuit_data.circuit
+            circuit.append(cirq.measure(circuit.all_qubits()))
+
+            noisy_results = cast(
+                MeasurementResult, self.executor._run([circuit])[0]
+            ).result
+            noisy_distribution = bitstrings_to_distribution(noisy_results)
+            ideal_results = (
+                cast(
+                    MeasurementResult, self.ideal_executor._run([circuit])[0]
+                ).result
+                if self.ideal_executor
+                else None
+            )
+            ideal_distribution = (
+                bitstrings_to_distribution(ideal_results)
+                if ideal_results
+                else {"1" * len(circuit.all_qubits()): 1.0}
+            )
+
+            mitigated: dict[str, dict[str, Any]] = {
+                technique: {"results": [], "method_improvement_factor": None}
+                for technique in self.settings.techniques
+            }
+            for strategy in self.settings.make_strategies():
+                most_probable_bitstring = max(
+                    ideal_distribution, key=ideal_distribution.get
+                )
+                expval_executor = bitstring_executor_to_expval_executor(
+                    self.executor, most_probable_bitstring
+                )
+                mitigated_expval = strategy.mitigation_function(
+                    circuit, expval_executor
+                )
+                mitigated[strategy.technique]["results"].append(
+                    {
+                        "circuit_type": circuit_data.type,
+                        "mitigated_value": mitigated_expval,
+                        **strategy.technique_params,
+                    }
+                )
+            expvals.append(
+                {
+                    "noisy_dist": noisy_distribution,
+                    "ideal_dist": ideal_distribution,
+                    "mitigated": mitigated,
+                }
+            )
+        return expvals
+
+    def compute_improvements(
+        self, experiment_results: list[dict[str, Any]]
+    ) -> None:
+        """Compute the improvement factors for each calibration circuit that
+        was run."""
+        for result in experiment_results:
+            ideal_dist = result["ideal_dist"]
+            ideal_expval = max(ideal_dist.values())
+            noisy_dist = result["noisy_dist"]
+            noisy_expval = max(noisy_dist.values())
+            for di in result["mitigated"].values():
+                results = di["results"]
+                mitigated_vals = list(
+                    map(lambda di: di["mitigated_value"], results)
+                )
+                method_improvement_factor = abs(
+                    noisy_expval - ideal_expval
+                ) / sqrt(
+                    len(mitigated_vals)
+                    * sum(
+                        (mitigated_val - ideal_expval) ** 2
+                        for mitigated_val in mitigated_vals
```

Comment:
```suggestion
                    noisy_value - ideal_value
                ) / sqrt(
                    len(mitigated_vals)
                    * sum(
                        (mitigated_val - ideal_value) ** 2
                        for mitigated_val in mitigated_vals
```

For consistency between `something_expval` vs `something_value`.
No preference between the two, but better to keep consistency.